crates/ra_syntax/src/ast/mod.rs

```diff
@@ -120,9 +120,17 @@ impl<'a> Comment<'a> {
     pub fn prefix(&self) -> &'static str {
         self.flavor().prefix()
     }
+
+    pub fn count_newlines_lazy(&self) -> impl Iterator<Item = &()> {
+        self.text().chars().filter(|&c| c == '\n').map(|_| &())
+    }
+
+    pub fn has_newlines(&self) -> bool {
+        self.count_newlines_lazy().count() > 0
+    }
```

Comment:
I'm not sure about duplicating these functions for `Comment` and `Whitespace`. Is there a better place to implement them?

Reply:
`SyntaxText` is probably the right place for them? We can also introduce a 

```
trait LeafNode: SyntaxNode {
    fn text(&self) -> &SmolStr
}
```

and implement it for `Whitespace` and `Comment`.